File: api/weather.py
import os
import requests
import cgi

key = os.environ.get('WEATHER_KEY')
form = cgi.FieldStorage()
city =  form.getvalue('searchbox')

get_weather()

def get_weather():
    query = {'q': city + ',US', 'units': 'imperial', 'appid': key}
    url = 'api.openweathermap.org/data/2.5/weather'
    try:
        data = requests.get(url, params=query).json()
        forecast = data['list']
    except KeyError:
        print(f'KeyError: Please check city name and country code in {query}')
        return 'Error! Please check city name and country code.', 0, 0, 0
    weather = forecast[0]
    description = weather['weather'][0]['description']
    temperature = weather['main']['temp']
    cloudy = weather['clouds']['all']
    humidity = weather['main']['humidity']
    description = description.capitalize()




    4c6a7c501ad077318a7c705eea755377

File: api/weather.js
const api = {
    key: '4c6a7c501ad077318a7c705eea755377',
    url: 'https://api.openweathermap.org/data/2.5/'
}

const city = document.querySelector('.city');
// const city = document.getElementById("city");
city.addEventListener('keypress', setQuery);
// const state = document.getElementById("state");


function setQuery(evt) {
    if (evt.keyCode == 13) {
        getResults(city.value);
        console.log(city.value);
    }
}

function getResults (query) {
    fetch(`${api.url}weather=?q=${query}&units=imperial&appid=${api.key}`)
    .then(weather => {
        return weather.json();
    }).then(displayResults);
}

function displayResults (weather) {
    let city = document.querySelector('.location .city');
    city.innerHTML = `${weather.name}, ${weather.sys.country}`;

    let temp = document.querySelector('.current .temp');
    temp.innerHTML = `${Math.round(weather.main.temp)}<span>*c</span>`;

    let weather_el = document.querySelector('current .weather');
    weather_el.innerText = weather.weather[0].main;

    let hilow = document.querySelector('.hilow');
    hilow.innerText = `${Math.round(weather.main.temp_min)}*c / ${Math.round(weather.main.temp_max)}*c`;
}